mirdata/giantsteps_key.py

```diff
@@ -0,0 +1,322 @@
+    #     with zipfile.ZipFile(os.path.join(data_home, 'audio.zip'), 'r') as zip_ref:
+    #         zip_ref.extractall(data_home)
+    # if 'metadata' in partial_download:
+    #     with zipfile.ZipFile(os.path.join(data_home, 'metadata.zip'), 'r') as zip_ref:
+    #         zip_ref.extractall(data_home)
+    # if 'key' in partial_download:
+    #     with zipfile.ZipFile(os.path.join(data_home, 'key.zip'), 'r') as zip_ref:
+    #         zip_ref.extractall(data_home)
+
+
+def validate(data_home=None, silence=False):
+    """Validate if a local version of this dataset is consistent
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths where the expected file exists locally
+            but has a different checksum than the reference
+
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    missing_files, invalid_checksums = utils.validator(
+        DATA.index, data_home, silence=silence
+    )
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    """Get the list of track IDs for this dataset
+
+    Returns:
+        (list): A list of track ids
+    """
+    return list(DATA.index.keys())
+
+
+def load(data_home=None):
+    """Load giantsteps_key dataset
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Returns:
+        (dict): {`track_id`: track data}
+
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    beatles_data = {}
+    for key in track_ids():
+        beatles_data[key] = Track(key, data_home=data_home)
+    return beatles_data
+
+
+def load_metadata(metadata_path):
+    """Load giantsteps_key format metadata data from a file
+
+    Args:
+        metadata_path (str): path to metadata annotation file
+
+    Returns:
+        (dict): loaded metadata data
+
+    """
+    if metadata_path is None:
+        return None
+
+    if not os.path.exists(metadata_path):
+        raise IOError("metadata_path {} does not exist".format(metadata_path))
+
+    with open(metadata_path) as json_file:
+        meta = json.load(json_file)
+    return meta
+
+
+def load_key(keys_path):
+    """Load giantsteps_key format key data from a file
+
+    Args:
+        keys_path (str): path to key annotation file
+
+    Returns:
+        (str): loaded key data
+
+    """
+    if keys_path is None:
+        return None
+
+    if not os.path.exists(keys_path):
+        raise IOError("keys_path {} does not exist".format(keys_path))
+
+    with open(keys_path) as f:
+        key = f.readline()
+
+    return key
```

Comment:
Can you have this to load a `utils.KeyData` object?

Reply:
We speak about it in this issue https://github.com/mir-dataset-loaders/mirdata/issues/268. Your data structure is for local keys (e.g. in the Beatles dataset). However in this dataset the key annotation is global (the main key of the track). That's why is redundant the utils.KeyData start_time and end_time.  
I think that mixing the local key structure and the global key structure can be confusing.

Reply:
Hmm I agree that it's a bit confusing. If it's one global key though, what about annotating it like `KeyData('start_times'=[0], 'end_times'=track_duration, 'keys'=[key])`? Perhaps a bit overkill, but we're trying to strike the balance between standardizing things and keeping them flexible. Soon, we're thinking about ways to make multi-dataset loaders for a particular task, and having some common data structures will help with that.

Reply:
Hello @rabitt, We are also implementing a loader for the [Mridangam Stroke Dataset](https://zenodo.org/record/1265188#.X3RP8tYp5mA) and we are having the same issue here. This dataset contains single strokes of the Mridangam (which is a percussion instrument) in different tonics, so we have a single tonic per stroke and consequently, per track. For that reason, to avoid this redundancy and make things more clear, maybe it would be nice to have a global key structure. But please, discuss it if you want and let us know the way you think is more convenient for the project, as we will try to use the decided method for all these datasets. Thanks!!